ophyd/tests/test_bluesky_protocols.py

```diff
@@ -1,93 +1,156 @@
+import subprocess
+import tempfile
+from typing import Callable, List, Literal
+
+import pytest
 from mypy import api
 
 imports = ""
 imports += "import ophyd;"
 imports += "from ophyd import flyers, sim;"
 imports += "hw = sim.hw();"
 
-
-def run(cmd):
-    normal_report, error_report, exit_status = api.run(
-        ["--command", cmd, "--follow-imports=skip"]
-    )
-    print(cmd.split(";")[-1])
-    print("  ", normal_report)
-    print("  ", error_report)
-    assert exit_status == 0
-
-
-def test_checkable():
+SupportedTypeCheckers = Literal["pyright", "mypy"]
+enabled_type_checkers: List[SupportedTypeCheckers] = ["pyright", "mypy"]
+
+
+@pytest.fixture(params=enabled_type_checkers, scope="session")
+def run(request) -> Callable[[str], None]:
```

Comment:
Not relevant to this PR, but it feels a little strange to have this test at all when we could just have a file which imports and defines these that a mypy CI run would pick up on.

Reply:
Very good point. I assumed there was a reason it was done that way and didn't want to change too much. Happy to replace this with a type checking step in the CI pipeline if that's the way people want to go.

Reply:
I suppose it's to mimick type checking downstream which could be using either.